com.unity.ml-agents.extensions/Runtime/Match3/Match3Sensor.cs

```diff
@@ -119,22 +129,13 @@ public int Write(ObservationWriter writer)
                 {
                     for (var c = 0; c < m_Columns; c++)
                     {
-                        var val = m_Board.GetCellType(r, c);
-                        for (var i = 0; i < m_NumCellTypes; i++)
+                        var val = m_GridValues(r, c);
+
+                        for (var i = 0; i < m_OneHotSize; i++)
```

Comment:
does this loop exist to 0 out the rest of one-hots?  Could it be simplified to something like:
```
writer[offset + val] = 1.0f;
```

Reply:
Yeah, as long as we know the memory is initialized to zero, we could do that (plus an extra check that `val < m_OneHotSize`)